adal/src/main/java/com/microsoft/aad/adal/AuthenticationContext.java

```diff
@@ -1278,6 +1278,10 @@ void deserialize(final String serializedBlob) throws AuthenticationException {
         // first, we'll fall back to passed in authority host and all the aliased hosts if necessary; If app receives the blob has the old
         // version of adal, since all the apps using token share library are using login.windows.net, token lookup will keep working.
         final TokenCacheItem tokenCacheItem = SSOStateSerializer.deserialize(serializedBlob);
+        if (StringExtensions.isNullOrBlank(tokenCacheItem.getAuthority()) ||
```

Comment:
Could tokenCacheItem be null?

Reply:
No, we throw an `AuthenticationException`  if we can't construct the TokenCacheItem. I have added a unit test case here for this specific case : https://github.com/AzureAD/azure-activedirectory-library-for-android/pull/1209/commits/678e24d37df8d44233ddb9e96a691f1a5c8ccbfc#diff-1f51ca351931fc74098edcfa59eecbd3R2630